src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -627,8 +627,9 @@ def get_kube_proxy_config(self) -> Union[Dict, ContainerServiceNetworkProfileKub
                 )
 
         # try to read the property value corresponding to the parameter from the `mc` object
-        if self.mc and self.mc.network_profile and self.mc.network_profile.kube_proxy_config is not None:
-            kube_proxy_config = self.mc.network_profile.kube_proxy_config
+        if self.decorator_mode == DecoratorMode.CREATE:
```

Comment:
confused by this checking of create won't self.mc.network_profile alsways be empty in create?

Reply:
network_profile I do not think is empty on create. for example we using 10.244.0.0/16 as the default pod_cidr from the sdk